src/browser/setupWorker/setupWorker.ts

```diff
@@ -177,11 +180,32 @@ export class SetupWorkerApi
       options,
     ) as SetupWorkerInternalContext['startOptions']
 
-    return await this.startHandler(this.context.startOptions, options)
+    this.workerActivationPromise = this.startHandler(
+      this.context.startOptions,
+      options,
+    )
+
+    this.workerActivationPromise.catch(() => {
+      this.context.state = 'cancelled'
```

Comment:
No matter who rejects the activation promise, always transition to the `cancelled` state.